delta-lake/common/src/main/scala/com/nvidia/spark/rapids/delta/GpuDeltaParquetFileFormatUtils.scala

```diff
@@ -98,28 +102,30 @@ object GpuDeltaParquetFileFormatUtils {
     closeOnExcept(rowIdxCol) { rowIdxCol =>
 
       val delVecCol = delVec.map { delVec =>
-        withResource(Scalar.fromBool(false)) { s =>
-          withResource(CudfColumnVector.fromScalar(s, batch.numRows())) { c =>
-            var table = new Table(c)
-            val posIter = new RoaringBitmapIterator(
-              delVec.getLongIteratorFrom(rowIdxStart),
-              rowIdxStart,
-              rowIdxStart + batch.numRows(),
-            ).grouped(Math.min(maxBatchSize, batch.numRows()))
-
-            for (posChunk <- posIter) {
-              withResource(CudfColumnVector.fromLongs(posChunk: _*)) { poses =>
-                withResource(Scalar.fromBool(true)) { s =>
-                  table = withResource(table) { _ =>
-                    Table.scatter(Array(s), poses, table)
+        delVectorScatterTimeMetric.ns {
+          withResource(Scalar.fromBool(false)) { s =>
+            withResource(CudfColumnVector.fromScalar(s, batch.numRows())) { c =>
+              var table = new Table(c)
```

Comment:
Nit: this table is created before it's needed, will leak if the calculation of posIter throws.  Can simply move this down to avoid.